physical_property/pKa/analysis/pKa_analysis.py

```diff
@@ -0,0 +1,1925 @@
+            dG = getG(msgroup_n1) - getG(msgroup_p0)
+            pka.pKa_bydG = (dG / C_unit)
+
+            macropkas.append(pka)
+
+    return macropkas
+
+
+
+
+
+
+# =============================================================================
+# MAIN
+# =============================================================================
+
+if __name__ == '__main__':
+
+    sns.set_style('whitegrid')
+    sns.set_context('paper')
+
+    # Read experimental data.
+    with open(EXPERIMENTAL_DATA_FILE_PATH, 'r') as f:
+        # experimental_data = pd.read_json(f, orient='index')
+        names = ('Molecule ID', 'pKa mean', 'pKa SEM',
+                 'Assay Type', 'Isomeric SMILES')
+        experimental_data = pd.read_csv(f, names=names, skiprows=1)
+
+    # Convert numeric values to dtype float.
+    for col in experimental_data.columns[1:7]:
+        experimental_data[col] = pd.to_numeric(experimental_data[col], errors='coerce')
+
+
+    experimental_data.set_index("Molecule ID", inplace=True)
+    experimental_data["Molecule ID"] = experimental_data.index
+    #print("Experimental data: \n", experimental_data)
+
+    # Import user map.
+    with open(USER_MAP_FILE_PATH, 'r') as f:
+        user_map = pd.read_csv(f)
+
+
+    # Configuration: statistics to compute.
+    stats_funcs = collections.OrderedDict([
+        ('RMSE', rmse),
+        ('MAE', mae),
+        ('ME', me),
+        ('R2', r2),
+        ('m', slope),
+        ('kendall_tau', kendall_tau)
+    ])
+    ordering_functions = {
+        'ME': lambda x: abs(x),
+        'R2': lambda x: -x,
+        'm': lambda x: abs(1 - x),
+        'kendall_tau': lambda x: -x
+    }
+    latex_header_conversions = {
+        'R2': 'R$^2$',
+        'RMSE': 'RMSE',
+        'MAE': 'MAE',
+        'ME': 'ME',
+        'kendall_tau': '$\\tau$'
+    }
+
+
+
+
+
+    # ======================================================================================================================
+    # TO DO:  Convert relative microstate free energies to microscopic pKas, then calculate MACRO pKas from the micro pKas
+    # ======================================================================================================================
+
+    # Load submissions data.
+    submissions_RFE = load_submissions(pKa_SUBMISSIONS_DIR_PATH, user_map)
+
+    # Loop through each submission, convert relative FE's to micro pKas, then convert to macro pKas
+    for sub in submissions_RFE:
+        print("Macro-pKa conversion %s" % sub.participant)
+        print("Molecule From  To  pKa(titr)  pKa(dG)")
+        # Compute macro pKa by two methods:
+        # 1. simulated titration
+        # 2. delta G method. Based on canonical ensemble of microstate groups that share the same formal charge.
+        macropkas = get_macropka(sub.data)
+        for pka in macropkas:
+            print("%6s   %2d   %2d    %8.3f %8.3f" %(pka.molecule, pka.transition_from, pka.transition_to, pka.pKa_bytitration, pka.pKa_bytitration))
```

Comment:
There is a typo here, it's comparing the titration method to itself. When the titration method is compared to the delta G method, they agree for MOST of the predictions, but there are still a couple of predictions that disagree with one another. 

This
`print("%6s   %2d   %2d    %8.3f %8.3f" %(pka.molecule, pka.transition_from, pka.transition_to, pka.pKa_bytitration, pka.pKa_bytitration))`

Should be 
`print("%6s   %2d   %2d    %8.3f %8.3f" %(pka.molecule, pka.transition_from, pka.transition_to, pka.pKa_bytitration, pka.pka.pKa_bydG))`

Reply:
An example:

- CBDD Group:
         - SM43 -- titration method = 13.903 , dG method = -13.903
         - SM32 -- titration method = 9.419 , dG method = 11.841